### ISSUE TITLE
Tests failing on Python 3.8

### ISSUE THREAD

**Issue by jaraco** [python/importlib_metadata#100]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 02:16

Since the release of Python 3.8, the tests have started failing, even though they were passing on early releases of Python 3.8.

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 02:18

Re-running the [pipelines against master](https://gitlab.com/python-devs/importlib_metadata/-/jobs/365481350), which passed tests when 0.24 was released, there are now two tests failing that passed before. This issue is to address the emergent failure on Python 3.8.

```
py38 run-test: commands[0] | python -m unittest discover
132 ..........................F....................
133 ======================================================================
134 FAIL: test_package_discovery (importlib_metadata.tests.test_main.DiscoveryTests)
135 ----------------------------------------------------------------------
136 Traceback (most recent call last):
137   File "/builds/python-devs/importlib_metadata/importlib_metadata/tests/test_main.py", line 160, in test_package_discovery
138     assert all(
139 AssertionError
140 ----------------------------------------------------------------------
141 Ran 47 tests in 0.286s
142 FAILED (failures=1)
143 ERROR: InvocationError for command /builds/python-devs/importlib_metadata/.tox/py38/bin/python -m unittest discover (exited with code 1)
```

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 02:35

The failed assertion is probably another manifestation of #91 - because `Distribution` is defined in two places.

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 02:47

mentioned in commit 255b695e062e369a6af16bf76243e476afc36a1b

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 09:32

changed title from **{-test_discovery-} failing on Python 3.8** to **{+Tests+} failing on Python 3.8**

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 09:32

changed the description

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 09:33

changed the description

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 09:33

It's worse than I'd imagined. The [coverage tests are also failing](https://gitlab.com/python-devs/importlib_metadata/-/jobs/365485590), due to the stdlib functionality taking precedence. I think a solution that also addresses #91 may be necessary to address this issue.

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 09:50

mentioned in commit 73427ca5595a4b7bf426dfbbfee658b0140228e0

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 10:10

mentioned in merge request !98

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 19:29

closed via merge request !98

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 19:29

mentioned in commit c854e9a5e707e3209787d440935862a418227896

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Nov 30, 2019, 19:29

closed via commit 73427ca5595a4b7bf426dfbbfee658b0140228e0
